classes/data-stores/ActionScheduler_DBStore.php

```diff
@@ -305,9 +307,42 @@ protected function get_query_actions_sql( array $query, $select_or_count = 'sele
 			$sql          .= " AND a.hook=%s";
 			$sql_params[] = $query[ 'hook' ];
 		}
+
 		if ( ! is_null( $query[ 'args' ] ) ) {
-			$sql          .= " AND a.args=%s";
-			$sql_params[] = $this->get_args_for_query( $query[ 'args' ] );
+			if( isset($query[ 'args_partial_match' ]) && $query[ 'args_partial_match' ] === true && $mysql_version >= '5.7' ) {
```

Comment:
1. > When referring to array items, only include a space around the index if it is a variable.
1. > Yoda Conditions
```suggestion
			if ( isset( $query['args_partial_match'] ) && true === $query['args_partial_match'] && $mysql_version >= '5.7' ) {
```